gdrdrv/gdrdrv.c

```diff
@@ -479,16 +566,20 @@ static int gdrdrv_pin_buffer(gdr_info_t *info, void __user *_params)
         }
     }
 
-
     // here a typical driver would use the page_table to fill in some HW
     // DMA data structure
 
-    params.handle = mr->handle;
-
     mutex_lock(&info->lock);
     list_add(&mr->node, &info->mr_list);
+
+    if (gdr_generate_mr_handle(info, mr) != 0) {
```

Comment:
why is this call placed here? it returns an error but keeps the mr in the list.
it should add mr to the list only when it is ok.